std/typecons.d

```diff
@@ -292,6 +296,24 @@ private:
     assert(u1.isEmpty);
 }
 
+@safe unittest
+{
+    int i;
+    struct S
+    {
+        ~this() @trusted
+        {
+            // context pointer doesn't exist when GC calls
+            if (&i > cast(void*) 1024)
```

Comment:
Is there a better way of checking for this?

Reply:
why is this needed?

Reply:
destroy calls the destructor first, before nullifying the context pointer. Then when the GC calls the destructor it segfaults when accessing `i` to increment it, because the context pointer is null.

Reply:
This works for me:
```suggestion
            if (this !is typeof(this).init)
```

Reply:
OK. I asked on the learn forum for a way that works well in general and @aG0aep6G suggested using `tupleof[$-1]`, so I've used `tupleof[0]` as `tupleof.length` here is 1.